src/main/java/uk/gov/hmcts/ccd/domain/model/casedataaccesscontrol/mapper/RoleAssignmentToAccessProfileMapperImpl.java

```diff
@@ -0,0 +1,89 @@
+
+@Component
+public class RoleAssignmentToAccessProfileMapperImpl implements RoleAssignmentToAccessProfileMapper {
+
+    @Override
+    public List<AccessProfile> toAccessProfiles(RoleAssignmentFilteringResult filteringResult,
+                                                CaseTypeDefinition caseTypeDefinition) {
+        List<RoleAssignment> roleAssignments = extractRoleAssignments(filteringResult);
+
+        if (hasGrantTypeExcluded(roleAssignments)) {
+            roleAssignments = filterRoleAssignments(roleAssignments);
+        }
+
+        List<AccessProfile> accessProfiles = new ArrayList<>();
+        for (RoleAssignment roleAssignment : roleAssignments) {
+            for (RoleToAccessProfileDefinition roleToAccessProfile : caseTypeDefinition.getRoleToAccessProfiles()) {
+                if (validRoleAssignments(roleToAccessProfile, roleAssignment)) {
+
+                    List<String> authorisationsList = Arrays.asList(roleToAccessProfile.getAuthorisation().split(","));
+
+                    List<String> roleAssignmentAuthorisation = roleAssignment.getAuthorisations();
+
+                    if (roleAssignmentAuthorisation != null && roleAssignmentAuthorisation.size() > 0) {
+                        for (String authorisation : authorisationsList) {
+                            if (roleAssignmentAuthorisation.contains(authorisation)) {
+                                accessProfiles.add(createAccessProfile(roleAssignment, roleToAccessProfile));
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        return accessProfiles;
+    }
+
+    private boolean validRoleAssignments(RoleToAccessProfileDefinition roleToAccessProfileDefinition,
+                                         RoleAssignment roleAssignment) {
+        return !roleToAccessProfileDefinition.getDisabled()
+            && roleAssignment.getRoleName().equals(roleToAccessProfileDefinition.getRoleName())
+            && !StringUtils.isEmpty(roleToAccessProfileDefinition.getAuthorisation());
+    }
+
+    private List<RoleAssignment> extractRoleAssignments(RoleAssignmentFilteringResult filteringResult) {
+        return filteringResult.getRoleAssignmentRoleMatchingResults()
+            .stream()
+            .map(pair -> pair.getLeft())
+            .collect(Collectors.toList());
+    }
+
+    private AccessProfile createAccessProfile(RoleAssignment roleAssignment,
+                                              RoleToAccessProfileDefinition roleToAccessProfile) {
+        AccessProfile accessProfile = new AccessProfile();
+
+        accessProfile.setReadOnly(roleToAccessProfile.getReadOnly() || roleAssignment.getReadOnly());
+        accessProfile.setClassification(roleAssignment.getClassification());
+
+        String caseTypeAccessProfiles = roleToAccessProfile.getAccessProfiles();
+        accessProfile.setAccessProfiles(Arrays
+            .asList(caseTypeAccessProfiles.split(",")));
```

Comment:
don't think this is correct. I think we should create on instance of `AccessProfile` for each access profile in the mapping access profiles list